File: Back-end/Service_LichSuGiaoHang/src/main/java/com/example/Service_LichSuGiaoHang/Model/LichSuGiaoHang.java
package com.example.Service_LichSuGiaoHang.Model;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "v_lichsugiaohang")
public class LichSuGiaoHang {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="madh")
	private Integer maDH;
	
	@Column(name="mashipper")
	private Integer maShipper;
	
	@Column(name="madvvc")
	private Integer maDVVC;
	
	@Column(name="tenkh")
	private String tenKH;
	
	@Column(name="diachi")
	private String diaChi;
	
	
	@Column(name="sdt_khachhang")
	private String sdt;
	
	@Column(name="tenshipper")
	private String tenShipper;
	
	@Column(name="tendvvc")
	private String tenDVVC;
	
	@Column(name="ngaylap")
	private String ngayLap;
	
	@Column(name="trangthai")
	private String trangThai;

	@Column(name="hinhthucthanhtoan")
	private String hinhThucThanhToan;
	
	@Column(name="tongtien")
	private Integer tongTien;

	

	public LichSuGiaoHang() {}
	

	
	public Integer getMaDH() { return maDH;}
	public void setMaDH(Integer mdh) { this.maDH = mdh; }
	
	public Integer getMaShipper() { return maShipper;}
	public void setMaShipper(Integer tsp) { this.maShipper = tsp; }
	
	public Integer getMaDVVC() { return maDVVC;}
	public void setMaDVVC(Integer tsp) { this.maDVVC = tsp; }
	
	public String getTenKH() { return tenKH;}
	public void setTenKH(String tl) { this.tenKH = tl; }
	
	
	public String getDiaChi() { return diaChi;}
	public void setDiaChi(String tl) { this.diaChi = tl; }
	
	public String getSDT() { return sdt;}
	public void setSDT(String sdt) { this.sdt = sdt; }

	public String getTenShipper() { return tenShipper;}
	public void setTenShipper(String ts) { this.tenShipper = ts; }
	
	public String getTenDVVC() { return tenDVVC;}
	public void setTenDVVC(String tl) { this.tenDVVC = tl; }
	
	public String getNgayLap() { return ngayLap;}
	public void setNgayLap(String tl) { this.ngayLap = tl; }
	
	public String getTrangThai() { return trangThai;}
	public void setTrangThai(String ml) { this.trangThai = ml; }
	
	public String getHinhThucThanhToan() { return hinhThucThanhToan;}
	public void setHinhThucThanhToan(String ml) { this.hinhThucThanhToan = ml; }
	
	public Integer getTongTien() { return tongTien;}
	public void setTongTien(Integer ml) { this.tongTien = ml; }
	
}

File: Front-end/UDPT_ChoOnline/model/LichSuGiaoHang.php
<?php
declare(strict_types=1);
require_once("./Model/TaiKhoan.php");
use Tobscure\JsonApi\Parameters;

require_once("./Model/CallAPI.php");

    class LichSuGiaoHangModel
    {
        public $MaDH;
        public $MaShipper;
        public $TenKH;
        public $DiaChi;
        public $TenShipper;
        public $SDTKhachHang;
        public $TenDVVC;
        public $NgayLap;
        public $TrangThai;
        public $HinhThucThanhToan;
        public $TongTien;

        function __construct() 
        {
            $this->MaDH = "";
            $this->MaShipper = "";
            $this->TenKH = "";
            $this->DiaChi = "";
            $this->TenShipper = "";
            $this->SDTKhachHang = "";
            $this->TenDVVC = "";
            $this->NgayLap = "";
            $this->TrangThai = "";
            $this->HinhThucThanhToan = "";
            $this->TongTien = "";
        }

        public static function deliveryhistory($madvvc)
        {
            $url = "http://localhost:8053/api/lichsugiaohang/" . $madvvc;

            $get_data = callAPI('GET', $url, false);
            $response = json_decode($get_data, true);
            
            $ds = array();

            if ($response) 
            {            
                foreach ($response as $row) {
                    $p = new LichSuGiaoHangModel();
                    $p->MaDH = $row["maDH"];
                    $p->MaShipper = $row["maShipper"];
                    $p->TenKH = $row["tenKH"];
                    $p->DiaChi = $row["diaChi"];
                    $p->TenShipper = $row["tenShipper"];
                    $p->SDTKhachHang = $row["sdt"];
                    $p->TenDVVC = $row["tenDVVC"];
                    $p->NgayLap = $row["ngayLap"];
                    $p->TrangThai = $row["trangThai"];
                    $p->HinhThucThanhToan = $row["hinhThucThanhToan"];                 
                    $p->TongTien = $row["tongTien"];
                    $ds[] = $p;
                }
            }

            return $ds;
        }

     
    }

?>